tendermint/src/trust_threshold.rs

```diff
@@ -73,3 +113,74 @@ impl Display for TrustThresholdFraction {
         write!(f, "{}/{}", self.numerator, self.denominator)
     }
 }
+
+/// Facilitates validation of [`TrustThresholdFraction`] instances when
+/// deserializing them.
+#[derive(Serialize, Deserialize)]
+pub struct RawTrustThresholdFraction {
+    #[serde(with = "serializers::from_str")]
+    numerator: u64,
+    #[serde(with = "serializers::from_str")]
+    denominator: u64,
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
Thanks @thanethomson! Beautifully done! 💯👌

Reply:
Cool! Wasn't sure if proptest was a bit overkill for this, but I think it's still worthwhile to have more of this kind of testing in the codebase.